app/truckProfile.truckCtrl.ts

```diff
@@ -0,0 +1,100 @@
+interface ITruckProfileTruckScope extends ng.IScope {
+    isOnline:boolean;
+    selectedTruck:ITruckProfile;
+    customMenuColors:CustomMenuColors;
+    activeTruck:IActiveTruck;
+    activeTrucks:Array<IActiveTruck>;
+    truckCoords:{latitude:number; longitude:number};
+    map:{center:{}; zoom:number};
+    mapHeight: string;
+    menu:IMenu;
+    icon:string;
+    coords:{latitude:number; longitude:number};
+
+    activeTruckCheck(activeTrucks:Array<IActiveTruck>, selectedTruckString:string);
+
+}
+angular.module('TruckMuncherApp').controller('truckProfileTruckCtrl', ['$scope', 'growl', '$stateParams', 'TruckProfileService', 'colorService', 'TruckService', 'MenuService', '$analytics', 'navigator',
+    ($scope, growl, $stateParams, TruckProfileService, ColorService, TruckService, MenuService, $analytics, navigator) => new TruckProfilePartialCtrl($scope, growl, $stateParams, TruckProfileService, ColorService, TruckService, MenuService, $analytics, navigator)]);
+
+class TruckProfilePartialCtrl {
+    constructor(private $scope:ITruckProfileTruckScope,
+                private growl:IGrowlService,
+                private $stateParams:ng.ui.IStateParamsService,
+                private TruckProfileService:ITruckProfileService,
+                private colorService:IColorService,
+                private TruckService:ITruckService,
+                private MenuService:IMenuService,
+                private $analytics:IAngularticsService,
+                private navigator:Navigator
+    ) {
+
+        $scope.selectedTruck = null;
+        $scope.map = {
+            center: {
+                latitude: 43.05,
+                longitude: -87.95
+            },
+            zoom: 13
+        };
+        $scope.mapHeight = screen.height / 4 + 'px';
+        $scope.truckCoords = {latitude: 0, longitude: 0};
+        $scope.icon = 'img/map_marker_green.png';
+        $scope.activeTrucks = [];
+        $scope.customMenuColors = null;
+
+        this.navigator.geolocation.getCurrentPosition(function (pos) {
```

Comment:
Do you need to get the current position before doing anything on this controller? Seems like an unnecessary delay. Location is really only needed if you can't find the truck they requested (for updating the profiles list) and grabbing the location of the truck if it's active.